benchto-driver/src/main/java/com/teradata/benchto/driver/execution/ExecutionSynchronizer.java

```diff
@@ -40,9 +52,34 @@
 
     private static final double GRAPHITE_WAIT_BETWEEN_REPORTING_RESOLUTION_COUNT = 2;
 
+    private static final Duration SHUTDOWN_ASYNC_TASKS_WAIT_TIMEOUT = Duration.ofMinutes(20);
+    private static final int SHUTDOWN_ASYNC_TASKS_WAIT_REPORT_TIMES = 20;
```

Comment:
Only if there are jobs running. Since Graphite will be queried with 90s delay and since the measurement loading may (in an edge case) take longer that benchmarking itself, this times might accumulate a bit.. That's why there is so considerable a timeout. But we still exit as soon as the last is done.